stdlib/public/Concurrency/Executor.swift

```diff
@@ -319,6 +461,192 @@ extension SerialExecutor {
 
 }
 
+@available(SwiftStdlib 6.2, *)
+extension SerialExecutor where Self: Equatable {
+
+  @available(SwiftStdlib 6.2, *)
+  public func isSameExclusiveExecutionContext(other: Self) -> Bool {
+    return self == other
+  }
```

Comment:
Are we changing in any way what is expected to be invoked when? If we're moving away from the using the returned executor then we may need to update the diagram:  https://github.com/swiftlang/swift-evolution/pull/2716/files#diff-a8714de4afecd9377aa16f6cc8ed9297e84a9c7b109feb6fe18246742c9e02b9